frontend/packages/dev-console/src/components/pipelines/PipelineVisualizationTask.tsx

```diff
@@ -81,22 +84,16 @@ export const PipelineVisualizationTask: React.FC<PipelineVisualizationTaskProp>
         name={task.name || ''}
         namespace={namespace}
         status={task.status || { duration: '', reason: 'pending' }}
+        isPipelineRun={!!task.status}
```

Comment:
You need to update this area `task.status` will be non-existent for both cases, tasks of a `pipeline` and `pending/cancelled` tasks of  a `pipelinerun`. So, it will not be a distinguishing factor of a `pipeline` vs `run` for all cases. It would be better to pass the `pipelineRun` as an optional prop to the `PipelineVisualizationGraph`. Using this `prop` you should be able to distinguish between `Pipeline` and `PipelineRun` Tasks as well as `Pending` and `Cancelled` task by putting it into the `Pipelinerunfilterreducer` functionality.

Reply:
now using the pipelineRunStatus to distinguish between pipeline and pipelineRun